libs/brig-types/src/Brig/Types/Team/LegalHold.hs

```diff
@@ -0,0 +1,316 @@
+    parseJSON = withObject "NewLegalHoldClient" $ \o ->
+        NewLegalHoldClient <$> o .:  "prekeys"
+                           <*> o .:  "last_prekey"
+
+-- This is the payload that the LH service expects
+data RequestNewLegalHoldClient = RequestNewLegalHoldClient
+    { userId :: !UserId
+    , teamId :: !TeamId
+    } deriving stock (Show, Eq, Generic)
+
+instance ToJSON RequestNewLegalHoldClient where
+    toJSON (RequestNewLegalHoldClient userId teamId) = object
+        $ "user_id"    .= userId
+        # "team_id"    .= teamId
+        # []
+
+instance FromJSON RequestNewLegalHoldClient where
+    parseJSON = withObject "RequestNewLegalHoldClient" $ \o ->
+        RequestNewLegalHoldClient <$> o .: "user_id"
+                                  <*> o .: "team_id"
+
+-- TODO: Review this datatype: the encoding/swagger could probably be
+--       very much simplified with a different type for ulhsrClientId
+data UserLegalHoldStatusResponse =
+    UserLegalHoldStatusResponse
+      { ulhsrStatus     :: UserLegalHoldStatus
+      , ulhsrLastPrekey :: Maybe LastPrekey -- ^ Exists only when status is Pending or Enabled
+      , ulhsrClientId   :: Maybe ClientId -- ^ Exists only when status is Pending or Enabled
+      }
+   deriving stock (Eq, Show, Generic)
+
+instance ToJSON UserLegalHoldStatusResponse where
+    toJSON (UserLegalHoldStatusResponse status lastPrekey' clientId') = object
+        $  "status"      .= status
+        #  "last_prekey" .= lastPrekey'
+        #  (if isJust clientId'
+              then "client" .= object ["id" .= clientId']
+              else (mempty, Null))
+        # []
+
+instance FromJSON UserLegalHoldStatusResponse where
+    parseJSON = withObject "UserLegalHoldStatusResponse" $ \o ->
+        UserLegalHoldStatusResponse <$> o .: "status"
+                                    <*> o .:? "last_prekey"
+                                    <*> ((o .:? "client") >>= maybe (return Nothing) (.:? "id"))
+
+data LegalHoldClientRequest =
+    LegalHoldClientRequest
+    { lhcrRequester  :: !UserId
+    , lhcrLastPrekey :: !LastPrekey
+    } deriving stock (Eq, Show, Generic)
+
+instance FromJSON LegalHoldClientRequest where
+  parseJSON = withObject "LegalHoldClientRequest" $ \o ->
+    LegalHoldClientRequest
+        <$> o .: "requester"
+        <*> o .: "last_prekey"
+
+instance ToJSON LegalHoldClientRequest where
+  toJSON (LegalHoldClientRequest requester lastPrekey') = object
+        $  "requester" .= requester
+        #  "last_prekey" .= lastPrekey'
+        # []
+
+-- Request body definition for the @/confirm@ endpoint on the LegalHold Service
+data LegalHoldServiceConfirm =
+    LegalHoldServiceConfirm
+    { lhcClientId     :: !ClientId
+    , lhcUserId       :: !UserId
+    , lhcTeamId       :: !TeamId
+    , lhcRefreshToken :: !Text -- ^ Replace with Legal Hold Token Type
+    } deriving stock (Eq, Show, Generic)
+
+-- TODO: Review this in a separate PR, requires sync'ing with LH team
```

Comment:
@dkovacevic can you verify that this looks good ?

Reply:
I am not sure I understand this code right but LH documentation is here:
https://services.zinfra.io/legalhold/swagger#!/default/confirm

Reply:
@dkovacevic or is your swagger out of date?